indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -608,13 +608,22 @@ private void changeTaskCountInIOConfig(int desiredActiveTaskCount)
     }
   }
 
+  /**
+   * Clears allocation information including active task groups, partition groups, partition offsets, and partition IDs.
+   * <p>
+   * Note: Does not clear {@link #pendingCompletionTaskGroups} so that the supervisor remembers that these
+   * tasks are publishing and auto-scaler does not repeatedly attempt a scale down until these tasks
+   * complete. If this is cleared, the next {@link #discoverTasks()} might add these tasks to
+   * {@link #activelyReadingTaskGroups}.
+   */
   private void clearAllocationInfo()
   {
     activelyReadingTaskGroups.clear();
     partitionGroups.clear();
     partitionOffsets.clear();
 
-    pendingCompletionTaskGroups.clear();
+    // Note: We intentionally do NOT clear pendingCompletionTaskGroups here.
```

Comment:
Since the original line of code has already been removed, this comment seems out of place here. It has already been called out in the javadoc anyway.